docker/scripts/functions.sh

```diff
@@ -403,6 +403,81 @@ function validate_bios_config() {
 	fi
 }
 
+# usage: bios_inventory $id $plan $facility
+function bios_inventory() {
+	local id=$1
+	local plan=$2
+	local facility=$3
+	local bios_json_fn="/tmp/bios.json"
+	local hollow_json_fn="/tmp/hollow.json"
+	local hollow_namespace="net.equinixplatform.bios"
+	local hollow_auth_url
+	local hollow_audience
+	local hollow_token
+	local hollow_response
+	local hollow_url
+
+	# When running the inventorybios command outside of the packet-hardware
+	# container, UTIL_RACADM7 and UTIL_SUM must be set to the locations of the
+	# racadm and sum binaries, respectively
+	if UTIL_RACADM7=/opt/dell/srvadmin/bin/idracadm7 UTIL_SUM=/opt/supermicro/sum/sum packet-hardware inventorybios --verbose -u localhost --dry --cache-file "${bios_json_fn}"; then
+		# Generate JSON with additional fields required by Hollow
+		if ! jq --null-input \
+			--arg ns "${hollow_namespace}" \
+			--arg id "${id}" \
+			--slurpfile data "${bios_json_fn}" \
+			'{Namespace: $ns, hardware_uuid: $id, data: $data[]}' >"${hollow_json_fn}"; then
+			echo "Warning: error while generating hollow json"
+			return 0
+		fi
+		echo "BIOS feature inventory JSON is:"
+		cat "${hollow_json_fn}"
+
+		# Disable tracing since we're handling secrets
+		set +x
+
+		# Write data to Hollow
+		if [[ -z ${HOLLOW_CLIENT_ID:-} ]] || [[ -z ${HOLLOW_CLIENT_REQUEST_SECRET:-} ]]; then
+			echo "Warning: env credentials for Hollow not found, not writing to Hollow"
+			set -x
+			return 0
+		fi
+
+		hollow_auth_url="https://auth.equinixmetal.com/oauth/token"
+		hollow_audience="https://hollow.platformequinix.net"
```

Comment:
why not declare	`local hollow_auth_url="..."` here instead of old-school-C-like split?
It would be nice to make these `local -r` for readonly, even though its not common in this code base but no time like the present to start! (or not)

Reply:
I've moved the local declarations down where shellcheck doesn't complain about them (`hollow_token` and `hollow_response` are not included as shellcheck complains that the local declaration might mask a return value).

I'm leaving off the `local -r` declarations, and feel it's better to do this consistently in one pass across the script(s) at a future time.